### DISCUSSION TITLE
Burgers equation tutorial

### DISCUSSION THREAD

**Discussion by miguelmoyers** [firedrakeproject/firedrake#2585] [Firedrake support]
Hello All,

I am a complete novice to Firedrake. I just installed it yesterday. I want to use it for one of my courses. I've been working through the tutorials and I ran into some issues. In the Burgers equation one, I changed the initial condition to ic = project(as_vector((sin(2*pi*x[0])+0.5*sin(pi*x[0]), 0)), V). For some reason, I get the absolute value of the function. Am I missing something? What is as_vector() doing? I've been at it for a couple of hours and cannot find anything online. Sorry if it is a very simple question.
Thank you for your help.

Miguel

**Comment by colinjcotter** [comment]
Hi Miguel,
  as_vector turns a list of expressions  into a UFL symbolic representation of a vector field with the list contents given the Cartesian components.

all the best
cjc

On 14 Oct 2022, at 07:42, miguelmoyers ***@***.***> wrote:

﻿

Hello All,

I am a complete novice to Firedrake. I just installed it yesterday. I want to use it for one of my courses. I've been working through the tutorials and I ran into some issues. In the Burgers equation one, I changed the initial condition to ic = project(as_vector((sin(2pix[0])+0.5sin(pix[0]), 0)), V). For some reason, I get the absolute value of the function. Am I missing something? What is as_vector() doing? I've been at it for a couple of hours and cannot find anything online. Sorry if it is a very simple question.
Thank you for your help.

Miguel

**Reply by miguelmoyers** [reply]
Hello Colin,

That is what I thought. Still, when I project the initial `condition` into V, it seems it is taking the absolute value of the function, any idea why?
I attach a plot of the initial condition and the code I am using (which is the one from the tutorial but with a different initial condition).

from firedrake import *
import matplotlib.pyplot as plt
n = 30
mesh = UnitSquareMesh(n, n)

V = VectorFunctionSpace(mesh, "CG", 2)
V_out = VectorFunctionSpace(mesh, "CG", 1)

u_ = Function(V, name="Velocity")
u = Function(V, name="VelocityNext")

v = TestFunction(V)

x= SpatialCoordinate(mesh)
ic = project(as_vector([sin(2*pi*x[0])+0.5*sin(pi*x[0]), 0]), V)

u_.assign(ic)
u.assign(ic)

nu = 0.00001

timestep = 1.0/n


F = (inner((u - u_)/timestep, v)
     + inner(dot(u,nabla_grad(u)), v) + nu*inner(grad(u), grad(v)))*dx

t = 0.0
end = 0.5
while (t <= end):
    solve(F == 0, u)
    u_.assign(u)
    t += timestep
    uforplot=project(u,V_out)
    trisurf(uforplot)
    plt.show()
    plt.close()
![burguresinit](https://user-images.githubusercontent.com/115775606/196009110-0160e632-b374-4cd6-919a-33f9f4ddd684.png)
